src/api/article/article.model.js

```diff
@@ -107,15 +111,16 @@ export const POPULATE_OPTIONS = {
   // TODO(uladbohdan): to merge with User basicFields.
   author: '-_id firstName lastName email role active bio imageUrl displayName',
   brand: '-_id slug names imageUrl imageUrlSmall',
-  collection: {
+  collection: publishedOnly => ({
     path: 'collectionId',
     select: '-_id name slug description imageUrl articles',
     populate: {
       path: 'articles',
+      match: publishedOnly ? { publishAt: { $lt: Date.now() } } : undefined,
```

Comment:
looks ugly

Reply:
it does. But that incapsulates all the logic inside of the only object, that's a nice thing